pennylane/math/multi_dispatch.py

```diff
@@ -769,12 +769,19 @@ def add(*args, like=None, **kwargs):
     """Add arguments element-wise."""
     if like == "scipy":
         return onp.add(*args, **kwargs)  # Dispatch scipy add to numpy backed specifically.
-    try:
-        return np.add(*args, **kwargs)
-    except TypeError:
-        # catch arg1 = torch, arg2=numpy error
-        # works fine with opposite order
-        return np.add(args[1], args[0], *args[2:], **kwargs)
+
+    if like == "torch":
+        # In autoray 0.6.5, np.add dispatches to torch instead of
+        # numpy if one parameter is a torch tensor and the other is
+        # a numpy array. torch.add raises an Exception if one of the
+        # arguments is a numpy array, so here we cast both arguments
+        # to be tensors.
+        dev = getattr(args[0], "device", None) or getattr(args[1], "device")

```

Comment:
is there ever a situation where args[0] doesn't have a device set, but args[1] does? we're concerned today with the opposite scenario, but I'm curious about why we need to check both, and why args[1] doesn't get a default option

Reply:
Hmm, I guess there isn't. If `torch.add` gets tensors on different devices, then it raises an error. Actually maybe it's better to keep that here than to put them on the same device.

Reply:
Oh never mind, the scenario I was thinking of is if either `args[0]` and `args[1]` is a NumPy array, in which case the `device` attribute isn't defined. We want to put the NumPy array on the same device as the torch tensor.